base_layer/wallet/src/transaction_service/service.rs

```diff
@@ -1329,11 +1328,17 @@ where
         transaction_broadcast_join_handles: &mut FuturesUnordered<
             JoinHandle<Result<TxId, TransactionServiceProtocolError<TxId>>>,
         >,
-        script: TariScript,
+        recipient_script: Option<TariScript>,
         payment_id: PaymentId,
     ) -> Result<TxId, TransactionServiceError> {
         let tx_id = TxId::new_random();
 
+        // If the script is not provided, use the default script as we will replace this later on here with a stealth
+        // one with the correct public key, for now we only care that the script size is correct
```

Comment:
```suggestion
        // For a stealth transaction, the script is not provided because the destination address is not known at this stage, 
        // which will be calculated using the stealth public key a bit later on. For now, we use the default script to ensure
        // that the script size is correct.
```

Reply:
We know the destination, we dont know the public key.

Reply:
// For a stealth transaction, the script is not provided because the public key that should be included is not known at this stage, 
        // which will be calculated using the stealth public key later on. For now, we include a default public key to ensure
        // that the script size is correct.